File: gfg/april2024/pyhton/12_sum_of_products.py
# CODE

class Solution:
    def pairAndSum(self, n, arr):
        res = 0
        for b in range(32):
            count = 0
            for i in range(n):
                if arr[i] & (1 << b):
                    count += 1
            res += (count * (count - 1) // 2) * (1 << b)
        return res

File: gfg/april2024/c++/12_sum_of_products.cpp
// CODE

class Solution {
    public:
    long long pairAndSum(int n, long long arr[]) {
        long long res = 0;
        for(int b = 0; b < 32; b++) {
            long long count = 0;
            for(int i = 0; i < n; i++) {
                if((arr[i] & (1L << b))) count++;
            }
            res += (count * (count - 1) / 2) * (1L << b);
        }
        return res;
    }
};